sign-up-task/src/Utils/validators.js

```diff
@@ -37,5 +39,32 @@ export const fieldValidator = (fieldName, value = '', passwordValue = '') => {
 
         default:
     }
+}
 
+export const signUpFormValidator = async (newUser, email) => {
+    let users = await signUpAPI.getUsers();
+    let isUser = users.some(user => user.email === email)
+    isUser && signUpAPI.setUser(newUser)
+    return isUser;
 }
+
+export const signInFormValidator = async (input, email, password) => {
+    let users = await signUpAPI.getUsers();
+    let user = emailValidator(users, email) || ""
+    let correctEmail = users.some(user => user.email === email)
+    let correctPassword = passwordValidator(user, password)
+
+    return {
+        email: correctEmail,
+        password: correctPassword
+    }
+}
+
+const emailValidator = (users, email) => {
+    let userToCheck;
+    users.forEach(user => user.email === email && (userToCheck = user))
+    return userToCheck;
+};
+
+
+const passwordValidator = (user, password) => (user.password === password)
```

Comment:
you better to bcrypt password before saving, and then check using bcrypt

Reply:
will try to bcrypt. I gave no experience now with it